File: Mobiles/ios/iOS-MobileO11yDemo/Features/RemoteActions/Models/ErrorSimulationType.swift
//
//  ErrorSimulationType.swift
//  iOS-MobileO11yDemo
//
//  Created for error simulation and observability testing
//

import Foundation

/// Represents the type of error simulation to apply for remote car actions.
/// This is used for testing and observability purposes.
enum ErrorSimulationType: String, CaseIterable, Identifiable {
    /// No error - normal operation
    case none
    
    /// Simulate a 400 Bad Request error
    case badRequest = "400"
    
    /// Simulate a 500 Internal Server Error
    case internalServerError = "500"
    
    /// Random error (15% probability) - default behavior
    case random
    
    var id: String { rawValue }
    
    /// Converts the enum to the header value format
    var headerValue: String {
        switch self {
        case .none:
            return "none"
        case .badRequest:
            return "400"
        case .internalServerError:
            return "500"
        case .random:
            return "random"
        }
    }
    
    /// User-friendly display name for the UI
    var displayName: String {
        switch self {
        case .none:
            return "No Error"
        case .badRequest:
            return "Bad Request (400)"
        case .internalServerError:
            return "Internal Server Error (500)"
        case .random:
            return "Random (15%)"
        }
    }
}

File: Mobiles/flutter/lib/features/car_actions/sub_features/remote_actions/domain/error_simulation_type.dart
/// Represents the type of error simulation to apply for remote car actions.
/// This is used for testing and observability purposes.
enum ErrorSimulationType {
  /// No error - normal operation
  none,

  /// Simulate a 400 Bad Request error
  badRequest,

  /// Simulate a 500 Internal Server Error
  internalServerError,

  /// Random error (15% probability) - default behavior
  random,
}

extension ErrorSimulationTypeX on ErrorSimulationType {
  /// Converts the enum to the header value format
  String toHeaderValue() {
    switch (this) {
      case ErrorSimulationType.none:
        return 'none';
      case ErrorSimulationType.badRequest:
        return '400';
      case ErrorSimulationType.internalServerError:
        return '500';
      case ErrorSimulationType.random:
        return 'random';
    }
  }

  /// User-friendly display name for the UI
  String get displayName {
    switch (this) {
      case ErrorSimulationType.none:
        return 'No Error';
      case ErrorSimulationType.badRequest:
        return 'Bad Request (400)';
      case ErrorSimulationType.internalServerError:
        return 'Internal Server Error (500)';
      case ErrorSimulationType.random:
        return 'Random (15%)';
    }
  }
}
